app/javascript/components/PostsContainer.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect, useCallback } from 'react'
+import PostsList from './PostsList'
+import TabGroup from './TabGroup'
+import ReactPaginate from 'react-paginate'
+import { saRequest } from '../utils/saRequest'
+
+export default function PostsContainer(props) {
+  const TAB_GROUP = [
+    { value: '1', label: 'posts' },
+    { value: '2', label: 'citations' },
+  ]
+  const [tabState, setTabState] = useState('1')
+  const [data, setData] = useState([])
+  const [pageCount, setPageCount] = useState()
+  const [page, setPage] = useState(1)
+  const [showPagination, setShowPagination] = useState(false)
+  const [hasPosts, setHasPosts] = useState(false)
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const getPostsFromPage = useCallback(() => {
+    const generateUrl = () => {
+      const type = props.tag ? 'tags' : 'users'
+      const id = props.tag ? props.tag.id : props.user.id
+      const tab = tabState === '1' ? 'posts' : 'citations'
+      return '/' + type + '/' + id + '/paginated_' + tab + '/' + page
+    }
+
+    let url = generateUrl()
+    saRequest
+      .get(url)
+      .set('accept', 'application/json')
+      .send(data)
+      .then((res) => {
+        console.log('res', res)
+        setData(res.body.posts)
```

Comment:
I think this should have error handling for when one of these things returns malformed response.